apps/linode-marketplace-kali-linux/roles/kali/tasks/vnc_setup.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Install XFCE and VNC packages
+  apt:
+    name:
+      - xfce4
+      - xfce4-goodies
+      - dbus-x11
+      - tigervnc-standalone-server
+      - expect
+    state: present
+
+- name: Create VNC user
+  user:
+    name: "{{ vnc_username }}"
+    password: "{{ vnc_password | password_hash('sha512') }}"
+    shell: /bin/bash
+
+- name: Set up VNC password
+  expect:
+    command: vncserver
+    responses:
+      Password: "{{ vnc_password }}"
+      Verify: "{{ vnc_password }}"
+      "Would you like to enter a view-only password (y/n)?": "n"
+  become: yes
+  become_user: "{{ vnc_username }}"
```

Comment:
This output to the log. Use no_log to avoid leaking secrets to the stackscript log.